scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,201 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("dtype", DTYPES)
```

Comment:
@swallan you could parameterize the trans values like:

```suggestion
@pytest.mark.parametrize("dtype,trans", 
                                            list(zip(DTYPES, ["N"]*len(DTYPES))) 
                                            + list(zip(REAL_DTYPES , ["T"]*len(REAL_DTYPES ))) 
                                            + list(zip(COMPLEX_DTYPES , ["H"]*len(COMPLEX_DTYPES ))))
```
or something like that.

Reply:
If you don't want to run the whole test with each transpose option, you could also do something like:
```suggestion
@pytest.mark.parametrize("dtype, trans", zip(DTYPES, [("N", "T")]*2 + [("N", "H")]*2)
```

Reply:
The first one seems to work pretty well, I like it.